cmake/configure_lvi_mitigation_build.cmake

```diff
@@ -6,12 +6,29 @@ function(configure_lvi_mitigation_build LVI_MITIGATION_BINDIR)
     if (NOT EXISTS "${LVI_MITIGATION_BINDIR}")
         message(FATAL_ERROR "${LVI_MITIGATION_BINDIR} does not exist.")
     endif()
-    # Perfer clang over gcc.
-    if (EXISTS "${LVI_MITIGATION_BINDIR}/clang-7")
-        set(CMAKE_C_COMPILER ${LVI_MITIGATION_BINDIR}/clang-7 PARENT_SCOPE)
-        set(CMAKE_CXX_COMPILER ${LVI_MITIGATION_BINDIR}/clang++-7 PARENT_SCOPE)
+
+    # Default to clang-7 unless CC or CXX environment variable is explicitly set.
+    set(C_COMPILER clang-7)
```

Comment:
Can this be the latest version of installed clang rather than clang-7?

Reply:
This script is used when building OE, which we explicitly set the requirement to clang-7. Do you think we need to add logic to differentiate the scenarios between building OE and apps?

Reply:
OE emits a warning if the clang version is not 7, but will still build. We should probably consider doing the same here.

Reply:
@anakrish  I've updated the PR such that when building the samples, the default will be clang instead of clang-7 (consistent to the make behavior). On the SDK build, clang-7 remains the default.

@jhand2 The warning is preserved with this PR.  Below is the example of specifying clang-8 when building OE.

`
CMake Warning at cmake/compiler_settings.cmake:8 (message):
  Open Enclave officially supports Clang 7 only, but your Clang version
  (8.0.0) is older or newer than that.  Build problems may occur.
`